sdk/eventgrid/Azure.Messaging.EventGrid/tests/CloudEventTests.cs

```diff
@@ -299,5 +299,44 @@ public async Task SerializesExpectedProperties_DerivedType()
 
             Assert.AreEqual(5, cloudEvent.Data.ToObjectFromJson<DerivedTestPayload>().DerivedProperty);
         }
+
+        [Test]
+        public async Task RespectsPortFromUriSendingCloudEvents()
+        {
+            var mockTransport = new MockTransport((request) =>
+            {
+                Assert.AreEqual(100, request.Uri.Port);
+                return new MockResponse(200);
+            });
+            var options = new EventGridPublisherClientOptions
+            {
+                Transport = mockTransport
+            };
+            EventGridPublisherClient client =
+               new EventGridPublisherClient(
+                   new Uri("https://contoso.com:100/api/events"),
+                   new AzureKeyCredential("fakeKey"),
+                   options);
+            var cloudEvent = new CloudEvent(
+                    "record",
+                    "Microsoft.MockPublisher.TestEvent",
+                    new DerivedTestPayload
+                    {
+                        Name = "name",
+                        Age = 10,
+                        DerivedProperty = 5
+                    });
+
+            Assert.AreEqual(5, cloudEvent.Data.ToObjectFromJson<DerivedTestPayload>().DerivedProperty);
+
+            List<CloudEvent> eventsList = new List<CloudEvent>()
+            {
+                cloudEvent
+            };
+
+            await client.SendEventsAsync(eventsList);
+
+            Assert.AreEqual(5, cloudEvent.Data.ToObjectFromJson<DerivedTestPayload>().DerivedProperty);
```

Comment:
Why are we asserting something that was asserted 2 lines before?

Reply:
Eh, we are running the cloudevents through the publisher which could mutate them with extensions.

Reply:
Seems like that should have it's own test.

Reply:
We do have tests for that.

Reply:
Then I would trim unrelated asserts from these ones.